### PR TITLE
Adjust age groups for alg2.

### LINKED ISSUE #572: BUG:  age groups for Bürgergeld
### Bug description

- GETTSIM has age limits implemented as [0-6], [7-13], [14-17], [18-24]
- they should be [0-5], [6-13], [14-17], [18-24]

### PR DESCRIPTION
**Pull request by cpz-research**
### What problem do you want to solve?

Closes #572 

### Todo

- [x] Implement changes.
- [x] New test cases
- [x] Document PR in CHANGES.md.

### CODE REVIEW

File: src/_gettsim/demographic_vars.py [hunk 1]
```diff
@@ -105,8 +106,26 @@ def kind_bis_15(alter: int, kind: bool) -> bool:
...
+    alter
+        See basic input variable :ref:`alter <alter>`.
+    kind
+        See basic input variable :ref:`kind <kind>`.
+
+    Returns
+    -------
+
+    """
+    out = kind and (alter < 18)
```
Comment (hmgaudecker) [inline src/_gettsim/demographic_vars.py:123]:
```suggestion
    out = kind and (alter <= 17)
```
Easier to match with function name.

File: src/_gettsim/parameters/arbeitsl_geld_2.yaml [hunk 2]
```diff
@@ -307,14 +307,14 @@ anteil_regelsatz:
   2005-01-01:
     zwei_erwachsene: 0.9
     weitere_erwachsene: 0.8
-    kinder_0_6: 0.6
-    kinder_7_13: 0.6
+    kinder_0_5: 0.6
```
Comment (hmgaudecker) [inline src/_gettsim/parameters/arbeitsl_geld_2.yaml:310]:
```suggestion
    kinder_bis_5: 0.6
```
Any reason we do not use the naming convention we have elsewhere? If easy to change, please do, else disregard.

File: src/_gettsim/parameters/arbeitsl_geld_2.yaml [hunk 3]
```diff
@@ -307,14 +307,14 @@ anteil_regelsatz:
...
+    kinder_6_13: 0.6
     kinder_14_24: 0.8
     reference: Artikel 1 G. v. 24.12.2003 BGBl. I S. 2954.
   2009-07-01:
     deviation_from: previous
     reference: Artikel 1 G. v. 02.03.2009 BGBl. I S. 416.
-    kinder_7_13: 0.7
+    kinder_6_13: 0.7
```
Comment (hmgaudecker) [inline src/_gettsim/parameters/arbeitsl_geld_2.yaml:317]:
```suggestion
    kinder_ab_6_bis_13: 0.7
```
Same, won't repeat.

File: src/_gettsim_tests/test_data/arbeitsl_geld_2/2019/hh_id_7.yaml
Comment (hmgaudecker) [inline src/_gettsim_tests/test_data/arbeitsl_geld_2/2019/hh_id_7.yaml:1]: File name should be the purpose of the test, `hh_id_x` came from the automatic conversion only.

`test_age_limits.yaml`

And add tests for other cutoffs, too (=bedarfsgemeinschaft members aged 13, 14, 17, 18, 24)

**Review by hmgaudecker — COMMENTED** [review]
Thanks! Just a couple of minor comments, should all be quick to implement. One might be slightly more work, feel free to disregard that one if that turns out to be the case.

File: src/_gettsim/demographic_vars.py [hunk 4]
```diff
@@ -13,10 +13,11 @@
     "anz_rentner_hh": {"source_col": "rentner", "aggr": "sum"},
     "anz_kinder_hh": {"source_col": "kind", "aggr": "sum"},
     "anz_kinder_tu": {"source_col": "kind", "aggr": "sum"},
-    "anz_kinder_bis_17_hh": {"source_col": "kind_bis_17", "aggr": "sum"},
+    "anz_kinder_bis_5_hh": {"source_col": "kind_bis_5", "aggr": "sum"},
     "anz_kinder_bis_6_hh": {"source_col": "kind_bis_6", "aggr": "sum"},
```
[the remark this answers is not in the document]
Comment (ChristianZimpelmann) [reply src/_gettsim/demographic_vars.py:17]: We still use it for `_arbeitsl_geld_2_alleinerz_mehrbedarf_m_hh`. And I think rightly so: 

> Bei Betreuung eines Kindes unter sieben oder zwei Kinder unter sechzehn Jahren beträgt der Mehrbedarf 36 Prozent.

[Source](https://buergergeld-antrag.org/alleinerziehende/)

[the remark this answers is not in the document]
Comment (hmgaudecker) [reply src/_gettsim/demographic_vars.py:17]: Had missed that somehow, apologies!

**Comment by ChristianZimpelmann** [comment]
Thanks for the Review! I incorporated all comments and added test cases for the other age thresholds.

**Review by hmgaudecker — APPROVED** [review]
Excellent, thank you!

**Pull request merged**